classes/plugins/PKPPubIdPluginHelper.inc.php

```diff
@@ -162,15 +162,20 @@ function execute($contextId, $form, $pubObject) {
 	 * @param $contextId integer
 	 * @param $form object
 	 * @param $pubObject object
+	 * @param $save boolean Whether the pub id shall be saved here
 	 * 	Submission, Representation, SubmissionFile + OJS Issue
 	 */
-	function assignPubId($contextId, $form, $pubObject) {
+	function assignPubId($contextId, $form, $pubObject, $save = false) {
 		$pubIdPlugins = PluginRegistry::loadCategory('pubIds', true, $contextId);
 		if (is_array($pubIdPlugins)) {
 			foreach ($pubIdPlugins as $pubIdPlugin) {
 				if ($form->getData($pubIdPlugin->getAssignFormFieldName())) {
 					$pubId = $pubIdPlugin->getPubId($pubObject);
-					$pubObject->setStoredPubId($pubIdPlugin->getPubIdType(), $pubId);
+					if ($save) {
```

Comment:
No, this was/is the default behaviour -- the pub object (e.g. Representation) is saved in the other form, and the pub id is only assigned to the object here prior to saving it in that other form.
Now, in OJS, using the "schedule for publication", this was the other way aroung -- first the article is saved (e.g. scheduled to an issue) and then the pub id is eventually assigned, thus in this case it has to be saved here.